src/main/scala/com/databricks/labs/overwatch/api/ApiMeta.scala

```diff
@@ -226,6 +227,17 @@ class ApiMetaFactory {
   }
 }
 
+trait CommonApiMethods {
```

Comment:
No need to create a trait, you can simply change the default implementation   
ApiMeta line 163
  private[overwatch] def getParallelAPIParams(jsonInput: Map[String, String]): Map[String, String] = {
    logger.log(Level.INFO, s"""Needs to be override for specific API for intializing Parallel API call function""")
    Map[String, String]()
  }

 to 

 def getParallelAPIParams(jsonInput: Map[String, String]): Map[String, String] = {
    Map(
      "start_value" -> s"""${jsonInput.get("start_value").get.toLong}""",
      "end_value" -> s"""${jsonInput.get("end_value").get.toLong}""",
      "increment_counter" -> s"""${jsonInput.get("increment_counter").get.toLong}""",
      "final_response_count" -> s"""${jsonInput.get("final_response_count").get.toLong}"""
    )
  }
}

Reply:
@sriram251-code  done